doc/contributing/distribution.md

```diff
@@ -25,3 +25,16 @@ only as a semver-major change, unless the related feature or project is
 documented as experimental. In addition, Node.js includes external projects as
 internal dependencies. These may be replaced or removed at any time, provided
 that doing so is not a breaking change.
+
+## Executables
+
+The Node.js distribution adds a number of executable commands into the user's
+`PATH`, such as `node`. The distribution will only include executables that run
+code that is vendored within the Node.js distribution. The distribution will not
+include executables to run code that is downloaded after installation.
```

Comment:
This distinction, even with the call out below, is unclear to me. I think I understand your intent, but the wording leaves a bit to the interpretation IMO.

Reply:
Can you suggest an alternative?

Reply:
I was trying to think of one but it is hard so I just hit send (sorry). Here are the thoughts I have on this wording and the confusion:

1. "downloaded after installation" could be interpreted in a bunch of ways. What you mean is partly defined below, but maybe there is better wording here but it is also escaping me.
2. The example of `foo` reads to me as saying "we would not have an executable which is a shim or wrapper" which is exactly what `corepack` does. Is this addition then intended to exclude `corepack` unless `corepack` moves away from the jumper binaries? I see the call out here in your comment for the PR, but that is not reflected in the content of this doc.
3. I don't think this wording provides a "framework for how to think about shipping new executable", and although something is certainly better than nothing if the wording is confusing then I am not sure how someone could read this and draw a conclusion on a new proposal (or an existing one).

So with that said, I think a good idea before nitpicking wording (I know that I did that in the worst way here with vague feedback, lol, sorry about that I will try to do better in the future) would be to outline some more concrete goals for this new addition.